modules/serialize/src/decode_body_header.ts

```diff
@@ -244,6 +244,13 @@ export function decodeNonFrameBodyHeader(
   // This will throw if the number is larger than Number.MAX_SAFE_INTEGER.
   // i.e. a 53 bit number
   const contentLength = contentLengthBN.toNumber()
+  /* Postcondition: Non-framed content length MUST NOT exceed AES-GCM safe limits.
+   * https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/data-format/message-body.md#encrypted-content-length
+   */
+  needs(
+    Maximum.BYTES_PER_AES_GCM_NONCE >= contentLength,
```

Comment:
```suggestion
    Maximum.BYTES_PER_AES_GCM_NONCE > contentLength,
```

The maximum is permissible. There may have been some confusion between the two checks in C?

https://github.com/aws/aws-encryption-sdk-c/blob/eb99852908f21f8ab7bf4567cd31a31a934d63a5/source/framefmt.c#L312

https://github.com/aws/aws-encryption-sdk-c/blob/eb99852908f21f8ab7bf4567cd31a31a934d63a5/source/framefmt.c#L208

I don't remember why the two checks are different or if this was intentional.

The spec allows max[1], Python allows max2], Java..I'm not finding a check, but I might be missing something.

[1] https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/data-format/message-body.md#encrypted-content-length
[2] https://github.com/aws/aws-encryption-sdk-python/blob/00c7e1dbfd0aade7ccb61e5e5735e26a8a239c38/src/aws_encryption_sdk/streaming_client.py#L408-L411